crates/bevy_reflect/src/path.rs

```diff
@@ -25,395 +35,911 @@ pub enum ReflectPathError<'a> {
     ExpectedStruct { index: usize },
     #[error("expected a list, but found a different reflect value")]
     ExpectedList { index: usize },
+    #[error("expected a struct variant, but found a different reflect value")]
+    ExpectedStructVariant { index: usize },
+    #[error("expected a tuple variant, but found a different reflect value")]
+    ExpectedTupleVariant { index: usize },
```

Comment:
should these maybe return the `VariantType` with the error?

Reply:
Good idea!